kernel/src/syscall.rs

```diff
@@ -64,7 +64,11 @@ pub enum Syscall {
     /// interrupts and callbacks.
     ///
     /// System call class ID 0
-    Yield,
+    Yield {
+        wait: bool,
+        terminate: bool,
+        address: *mut u8,
+    },
```

Comment:
Splitting the parsing of yield into two places (once when this enum is created, and then again in sched) makes this a bit difficult to follow. Following memop's lead, this could just store values, and leave the interpretation of those values when the syscall is handled.

Reply:
Makes sense. This two-place parsing was from when address was an Option.

Reply:
Addressed.